### ISSUE TITLE
Problems building CUETools from source code for debugging

### ISSUE THREAD

**Issue by RDamman** [gchudov/cuetools.net#117]
I have tried to build CUETools for debugging CUERipper to investigate the problem I have found. I already had Visual Studio 2017 CE installed. First I updated Visual Studio 2017. Then I followed the manual for building:

Installing from sources
    • Create a directory to put in the project
    • Get the CUETools sources from GitHub (https://github.com/gchudov/cuetools.net):
**git clone https://github.com/gchudov/cuetools.net.git** 
    • Change the directory to **cuetools.net** and get the required submodules using:
**git submodule update --init --recursive** 
    • Apply patches to submodules:
**git apply --directory=ThirdParty/flac ThirdParty/submodule_flac_CUETools.patch**
**git apply --directory=ThirdParty/WavPack ThirdParty/submodule_WavPack_CUETools_VS2017.patch** 
    • The solution can be built using Microsoft Visual Studio 2017 or newer (Community Edition will work) 
        ◦ Install the required .NET development tools (currently .NET Framework 4.7 and .NET Core 2.0) (Main menu => Tools => Get Tools and Features...)
        ◦ Install an appropriate Windows SDK version (currently 10.0.16299.0) (Main menu => Tools => Get Tools and Features… => Individual components)
        ◦ Install the Microsoft Visual Studio Installer Projects (Main menu => Tools => Extensions and Updates... => Online => Visual Studio Marketplace => Microsoft Visual Studio Installer Projects). Restart Visual Studio required after downloading for installation.
    • Optional: Install NASM and add it to your PATH. This is required for building the 32-bit flac plugin. So I have chosen the 32-bit NASM to install.
    • Open cuetools.net\CUETools\CUETools.sln I had to retarget the SDK version (In the _Solution Explorer_ windows right click for popup menu and select _Retarget solution_ or Main Menu => Project => Retarget Solution)
    • Select 'Any CPU' under 'Solution Platforms'. I haven chosen under (In the _Solution Explorer_ windows right click for popup menu and select _Configuration Manager_ ) as _Active Solution Platform_ Win32 for debugging.
    • Build solution (Main menu => Build => Build Solution): I got as result _========== Build: 7 succeeded, 3 failed, 0 up-to-date, 60 skipped ==========_. 
A lot of DLLs were not build. **Did I do something wrong?** I successively selected several projects in the soluction explorer to build these project DLLs manually. After that I tried to start CUERipper but got the message at the drive selection _"Failed to load riper module"_. A little bit of debugging showed that the code in CUETools.Processor tries to load the plugin DLLs from not existing directories when starting CUERipper.
I am not a regular user of Visual Studio. **What am I doing wrong?**

**Comment by c72578** [comment]
'Any CPU' has to be built first.

**Comment by RDamman** [comment]
> 
> 
> 'Any CPU' has to be built first.

Hallo Wolfgang,

Thank you for your reply.
I have rebuild the solution after switching to _AnyCPU_ in the configuration manager but the problem persists.
Apparently _if (CUEProcessorPlugins.ripper == null)_ at line 297 in _frmCUERipper.cs_. This variable ripper of type ICDRipper is set in line 52 in _CUEProcessorPlugins.cs_ where the modules are loaded. The interface _ICDRipper_ is only implemented by class _CDDriveReader_ at line 40 in _SCSIDrive.cs_ in project _CUETools.Ripper.SCSI_.  My conclusion is that the module _CUETools.Ripper.SCSI.dll_ is not found at startup.
 _CUETools.Ripper.SCSI.dll_ is found after rebuild with _AnyCPU_ at:
- .\cuetools.net\CUETools.Ripper.SCSI\obj\Debug\net47\CUETools.Ripper.SCSI.dll
- .\cuetools.net\bin\Debug\plugins\net47\CUETools.Ripper.SCSI.dll
- .\cuetools.net\CUETools.Ripper.SCSI\obj\Debug\net20\CUETools.Ripper.SCSI.dll
- .\cuetools.net\bin\Debug\plugins\net20\CUETools.Ripper.SCSI.dll

But CUERipper wants to load the plugins in _CUEProcessorPlugins.cs_ from .\cuetools.net\bin\Debug\\**net47\plugins\win32**.

**Do I have to move files manually?**

**Comment by c72578** [comment]
@RDamman Yes, currently the required dll files need to be copied manually to the expected location after a Debug build.
For now, I will add a script, which copies the dll files - similar to https://github.com/gchudov/cuetools.net/blob/master/CUETools/collect_files.bat, but for the Debug build dlls.
